freeipa_auth/backends.py

```diff
@@ -159,23 +170,25 @@ def update_user_groups(self, user, group_names, perm_codenames):
             old_perms = user.user_permissions.exclude(codename__in=perm_codenames)  # noqa: E501
             user.user_permissions.remove(*old_perms)
 
-    def get_django_user_groups(self, groups):
+    def get_django_user_groups(self, user_session):
         """
         Parses and returns django specific user groups from freeipa user groups
         :return:
         """
+        groups = self.get_all_user_groups(user_session)
```

Comment:
`get_all_user_groups` gets called twice in quick succession.  (once in get_django_user_groups, then in get_django_user_perms).  Not a huge deal, seems unlikely to be a bottleneck, but if there's a simple refactoring that keeps code clarity and avoids that duplication, could be nice :)

Reply:
doesn't hurt to change it honestly